autofaiss/external/quantize.py

```diff
@@ -284,6 +289,13 @@ def write_ids_df_to_parquet(ids: pd.DataFrame, batch_id: int):
             )
         if nb_indices_to_keep > 1:
             indices_folder = cast(str, indices_folder)
+            max_nb_threads = max(
```

Comment:
this might be good but note that using multiple threads here will impact the results: if nb_indices_to_keep is more than the number of available cores then the search will be slower and hence the efsearch and nprobe parameters will be adapted accordingly

Reply:
I think it's ok though as using multiple indices in parallel may be how the user want to use the N indices anyway

Reply:
I just saw this comment. It is indeed possible that the multiple threads impact the speed and thus the hyperparameters tuning. Maybe we should put number of processors into the max function. 
Do you know `binary_search_on_param` would benefit from multiple threads or it is using single thread ?

Reply:
The code makes it use a single core explicitly (using the omp set thread function)

Reply:
I created a PR to fix this: https://github.com/criteo/autofaiss/pull/97